src/editor/components/ErrorBoundary.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * ErrorBoundary component for 3DStreet that catches React errors while allowing
+ * the underlying A-Frame/Three.js canvas to continue running.
+ */
+class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+      error: null,
+      errorInfo: null,
+      isDetailsExpanded: false
+    };
+    this.copyErrorDetailsRef = React.createRef();
+  }
+
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    // Log the error to the console
+    console.error('Error caught by ErrorBoundary:', error, errorInfo);
+
+    // Update state with error details
+    this.setState({ errorInfo });
+
+    // Call the onError callback if provided
+    if (this.props.onError) {
+      this.props.onError(error, errorInfo);
+    }
+  }
+
+  toggleDetails = () => {
+    this.setState((prevState) => ({
+      isDetailsExpanded: !prevState.isDetailsExpanded
+    }));
+  };
+
+  copyErrorDetails = () => {
+    const { error, errorInfo } = this.state;
+    const errorDetails = `
+Error: ${error?.toString() || 'Unknown error'}
+
+Component Stack:
+${errorInfo?.componentStack || 'No component stack available'}
+
+URL: ${window.location.href}
+User Agent: ${navigator.userAgent}
+Date/Time: ${new Date().toISOString()}
+    `.trim();
+
+    try {
+      navigator.clipboard.writeText(errorDetails).then(
+        () => {
+          // Show a temporary success message
+          const copyButton = this.copyErrorDetailsRef.current;
+          if (copyButton) {
+            const originalText = copyButton.textContent;
+            copyButton.textContent = 'Copied!';
+            setTimeout(() => {
+              copyButton.textContent = originalText;
+            }, 2000);
+          }
+        },
+        (err) => {
+          console.error('Failed to copy error details:', err);
+        }
+      );
+    } catch (err) {
+      console.error('Copy to clipboard not supported:', err);
+
+      // Fallback for browsers that don't support clipboard API
+      const textarea = document.createElement('textarea');
+      textarea.value = errorDetails;
+      textarea.style.position = 'fixed';
+      document.body.appendChild(textarea);
+      textarea.focus();
+      textarea.select();
+
+      try {
+        document.execCommand('copy');
+        const copyButton = this.copyErrorDetailsRef.current;
+        if (copyButton) {
+          const originalText = copyButton.textContent;
+          copyButton.textContent = 'Copied!';
+          setTimeout(() => {
+            copyButton.textContent = originalText;
+          }, 2000);
+        }
+      } catch (e) {
+        console.error('Fallback copy failed:', e);
+      }
+
+      document.body.removeChild(textarea);
+    }
```

Comment:
The fallback is used for non-https and IE11.
navigator.clipboard.writeText should be implemented in all evergreen browsers.
https://caniuse.com/mdn-api_clipboard_writetext
Note that in the code base we use clipboard dependency to do the clipboard.writeText with fallback used in  
Component.js CommonComponents.js StreetSegmentComponent.js and I replaced that dependency by clipboard-copy that is more lightweight in aframe-inspector. https://www.npmjs.com/package/clipboard-copy
We also just use clipboard.writeText without fallback in AIChatPanel.js ScreenshotModal.component.jsx
For consistency, I would remove the fallback and remove the clipboard dependency, or use the clipboard-copy dependency everywhere.